source/ui/Components/editor/Editor.qml

```diff
@@ -563,70 +565,165 @@ ScrollView {
+                  propagateComposedEvents: false
+                  preventStealing: true
+                  hoverEnabled: true
+                  onClicked: {
+                      sidebar.addBreakpoint(Math.floor(mouse.y/textArea.cursorRectangle.height));
+                  }
+                }
+
+                // Displays line numbers.
+                Rectangle {
                     id: lineNumbersBar
+
                     anchors.left: parent.left
-                    anchors.leftMargin: 3
                     y: textArea.textMargin/2
-
-                    Repeater {
-                        model: textArea.lineCount
-                        delegate: Text {
-                            color: "gray"
-                            font: textArea.font
-                            text: {
-                                // Check if line number belongs to line which was inserted for a macro expansion.
-                                if (textArea.isPositionInsideMacroBlankLine(textArea.text, TextUtilities.getLineStartForLine(textArea.text, index))) {
-                                    return " ";
-                                } else {    // If not, return line number with blank lines factored out.
-                                    return textArea.convertRawLineNumberToDisplayLineNumber(textArea.text, index);
+                    width: fontMetrics.averageCharacterWidth * textArea.lineCount.toString().length
+                    anchors.leftMargin: 2
+
+                    Column {
+                        id: lineNumbers
+                        anchors.fill: parent
+
+                        Repeater {
+                            model: textArea.lineCount
+                            delegate: Text {
+                                color: "gray"
+                                font: textArea.font
+                                text: {
+                                    // Check if line number belongs to line which was inserted for a macro expansion.
+                                    if (textArea.isPositionInsideMacroBlankLine(textArea.text, TextUtilities.getLineStartForLine(textArea.text, index))) {
+                                        return " ";
+                                    } else {    // If not, return line number with blank lines factored out.
+                                        return textArea.convertRawLineNumberToDisplayLineNumber(textArea.text, index);
+                                    }
                                 }
+                                height: textArea.cursorRectangle.height
+
+                                horizontalAlignment: Text.AlignRight
+                                anchors.right: parent.right
                             }
-                            height: textArea.cursorRectangle.height
                         }
                     }
                 }
 
+                // Display errors, warnings and notes. Overlays lineNumberBar.

```

Comment:
Weren't the breakpoints and errors supposed to be in a bar next to the linenumbers in order to prevent line numbers from being obstructed? I actually fixed this once in a review request.